packages/async-flow/src/template.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * @import {HostCall} from './types.js';
+ */
+
+const { assign } = Object;
+
+/** Doesn't need to be exhaustive, just a little prettier than JSON-quoting. */
+const BEST_GUESS_ID_REGEX = /^[a-zA-Z_$][a-zA-Z0-9_$]*$/;
+
+/**
+ * Return an object that mimics a template strings array.
+ *
+ * @param {string[]} strings
+ * @returns {TemplateStringsArray}
+ */
+export const makeTemplateStringsArray = strings =>
+  harden(assign([...strings], { raw: strings }));
+harden(makeTemplateStringsArray);
+
+/**
+ * When used as a template tag, this function returns its arguments verbatim.
+ *
+ * @template {any[]} A
+ * @param  {A} allArgs
```

Comment:
IIUC, this is appropriate type notation for a single parameter. This needs the type notation for all the rest of the parameters, which I think also uses a `...`

Reply:
This use of `A` is the type of the rest arguments as a whole.  No more JSDoc annotation is necessary, because TS handles the `...allArgs` in the function declaration.  TS correctly types `allArgs` as `A`, which is an array or tuple as inferred or specified by the caller.

Reply:
surprised. nice to know